cli/dcoscli/marathon/main.py

```diff
@@ -32,11 +32,17 @@ def _main(argv):
         argv=argv,
         version='dcos-marathon version {}'.format(dcoscli.version))
 
-    return cmds.execute(_cmds(), args)
+    marathon_client = marathon.create_client()
+    pod_subcommand = MarathonPodSubcommand(resource_reader=_get_resource,
+                                           marathon_client=marathon_client)
```

Comment:
For all non-pod commands, marathon.create_client will happen twice, and pods command is a small subset of the marathon commands. I don't think initializing the class later is a big deal.